src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Windows.cs

```diff
@@ -35,8 +35,10 @@ internal bool TryGetCachedLength(out long cachedLength)
             return _lengthCanBeCached && cachedLength >= 0;
         }
 
-        internal static unsafe SafeFileHandle Open(string fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, long preallocationSize)
+        internal static unsafe SafeFileHandle Open(string fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, long preallocationSize, UnixFileMode? unixCreateMode = null)
         {
+            Debug.Assert(!unixCreateMode.HasValue);
```

Comment:
This method gets called from code that is common to Windows and Unix.
I've added an assert here because the caller should already have thrown PNSE when we get here.

We may want to split this code, see https://github.com/dotnet/runtime/pull/69980/files#r887771293, which may get rid of the argument here.

Reply:
What is the purpose of the argument? I don't see it being consumed inside the method, aside from the `Debug.Assert` check.

Reply:
The first comment explains it. This gets called from code that is common to Windows and Unix.